src/python/qgspythonutilsimpl.cpp

```diff
@@ -603,49 +603,61 @@ QStringList QgsPythonUtilsImpl::pluginList()
 QString QgsPythonUtilsImpl::getPluginMetadata( const QString &pluginName, const QString &function )
 {
   QString res;
-  QString str = "qgis.utils.pluginMetadata('" + pluginName + "', '" + function + "')";
+  QString str = QStringLiteral( "qgis.utils.pluginMetadata('%1', '%2')" ).arg( pluginName, function );
   evalString( str, res );
   //QgsDebugMsg("metadata "+pluginName+" - '"+function+"' = "+res);
   return res;
 }
 
+bool QgsPythonUtilsImpl::pluginHasProcessingProvider( const QString &pluginName )
+{
+  return getPluginMetadata( pluginName, QStringLiteral( "hasProcessingProvider" ) ).compare( QLatin1String( "yes" ), Qt::CaseInsensitive ) == 0;
```

Comment:
@nyalldawson I just noticed that in the `metadata.txt`, we have booleans for `experimental`, `deprecated` and `server`.

Your new parameter `hasProcessingProvider` is a `yes`/`no`. Can we change?